lib/amqp_client.js

```diff
@@ -378,11 +379,23 @@ AMQPClient.prototype.createReceiver = function(address, options, cb) {
   var linkPolicy = u.deepMerge({
     options: {
       name: linkName,
-      source: { address: address },
+      source: { address: address.name },
       target: { address: 'localhost' }
     }
   }, self.policy.receiverLink);
 
+  // if a subject has been provided then automatically set up a filter to match
+  // on that subject.
+  if (!!address.subject) {
+    var filterSymbol = (address.subject.indexOf('*') || address.subject.indexOf('#')) ?
+      'apache.org:legacy-amqp-topic-binding:string' :
```

Comment:
I guess my point is that this seems like it might be something that could be factored into a functor inside a qpid policy, with functors for the other policies that "did the right thing" with their "subjects"